hudi-common/src/main/scala/org/apache/hudi/BaseHoodieTableFileIndex.scala

```diff
@@ -51,19 +51,14 @@ import scala.language.implicitConversions
  * @param shouldIncludePendingCommits flags whether file-index should exclude any pending operations
  * @param fileStatusCache transient cache of fetched [[FileStatus]]es
  */
-abstract class HoodieTableFileIndexBase(engineContext: HoodieEngineContext,
+abstract class BaseHoodieTableFileIndex(engineContext: HoodieEngineContext,
```

Comment:
there is a large overlap between this and JavaBaseHoodieTableFileIndex right. do we plan to dedup in follow up patch ?

Reply:
Scala one will go away